tests-e2e/timestamp_test.go

```diff
@@ -0,0 +1,126 @@
+					TimelineLimit: 10,
+				},
+				BumpEventTypes: []string{"m.reaction"}, // only reactions bump the timestamp
+			},
+		},
+	})
+
+	// Bob should see a different timestamp than alice, as he just joined
+	bobTimestampBefore := resBob.Rooms[roomID].Timestamp
+	if bobTimestampBefore == timestampBeforeBobJoined {
+		t.Fatalf("expected timestamp to differ: %v vs %v", timestampBeforeBobJoined, bobTimestampBefore)
+	}
+
+	// Send an event which should NOT bump Bobs timestamp, because it is not listed it
+	// any BumpEventTypes
+	emptyStateKey := ""
+	eventID := alice.SendEventSynced(t, roomID, Event{
+		Type:     "m.room.topic",
+		StateKey: &emptyStateKey,
+		Content: map[string]interface{}{
+			"topic": "random topic",
+		},
+	})
+
+	bobTimestampAfter := resBob.Rooms[roomID].Timestamp
+
+	resBob = bob.SlidingSyncUntilEventID(t, resBob.Pos, roomID, eventID)
+	t.Logf("bobTimestampBefore vs bobTimestampAfter: %v %v", bobTimestampBefore, bobTimestampAfter)
+	if bobTimestampBefore != bobTimestampAfter {
+		t.Fatalf("expected timestamps to be the same, but they aren't: %v vs %v", bobTimestampBefore, bobTimestampAfter)
+	}
+	bobTimestampBefore = bobTimestampAfter
+
+	// Now send a message which bumps the timestamp in myFirstList
+	eventID = alice.SendEventSynced(t, roomID, Event{
+		Type: "m.room.message",
+		Content: map[string]interface{}{
+			"msgtype": "m.text",
+			"body":    "Hello, world!",
+		},
+	})
+
+	resBob = bob.SlidingSyncUntilEventID(t, resBob.Pos, roomID, eventID)
+	bobTimestampAfter = resBob.Rooms[roomID].Timestamp
+	if bobTimestampBefore == bobTimestampAfter {
+		t.Fatalf("expected timestamps to be different, but they aren't: %v vs %v", bobTimestampBefore, bobTimestampAfter)
+	}
+	bobTimestampBefore = bobTimestampAfter
+
+	// Now send a message which bumps the timestamp in mySecondList
+	eventID = alice.SendEventSynced(t, roomID, Event{
+		Type: "m.reaction",
+		Content: map[string]interface{}{
+			"m.relates.to": map[string]interface{}{
+				"event_id": eventID,
+				"key":      "✅",
+				"rel_type": "m.annotation",
+			},
+		},
+	})
+
+	resBob = bob.SlidingSyncUntilEventID(t, resBob.Pos, roomID, eventID)
+	bobTimestampAfter = resBob.Rooms[roomID].Timestamp
+	if bobTimestampBefore == bobTimestampAfter {
+		t.Fatalf("expected timestamps to be different, but they aren't: %v vs %v", bobTimestampBefore, bobTimestampAfter)
+	}
+	bobTimestampBefore = bobTimestampAfter
+
+	// Send another event which should NOT bump Bobs timestamp
+	eventID = alice.SendEventSynced(t, roomID, Event{
+		Type:     "m.room.name",
+		StateKey: &emptyStateKey,
+		Content: map[string]interface{}{
+			"name": "random name",
+		},
+	})
+
+	resBob = bob.SlidingSyncUntilEventID(t, resBob.Pos, roomID, eventID)
+	bobTimestampAfter = resBob.Rooms[roomID].Timestamp
+	if bobTimestampBefore != bobTimestampAfter {
+		t.Fatalf("expected timestamps to be the same, but they aren't: %v vs %v", bobTimestampBefore, bobTimestampAfter)
+	}
+}
```

Comment:
Needs more tests. Specifically:
- If you now do an initial sync (no pos) as Bob with the same lists, I don't think you'll get the right value. You'll get the latest timestamp (m.room.name) and not the m.reaction timestamp.
- Ensure you `time.Sleep(time.Millisecond)` after each send to ensure that timestamps will change.
- If Charlie now joins the room with the same bump_event_types as bob, he should see the timestamp of his join.